pex/dist_metadata.py

```diff
@@ -789,14 +789,20 @@ def as_constraint(self):
 class DistMetadata(object):
     @classmethod
     def from_metadata_files(cls, metadata_files):
-        # type: (MetadataFiles) -> DistMetadata
-        return cls(
-            files=metadata_files,
-            project_name=metadata_files.metadata.project_name,
-            version=metadata_files.metadata.version,
-            requires_dists=tuple(requires_dists(metadata_files)),
-            requires_python=requires_python(metadata_files),
-        )
+        project_name = metadata_files.metadata.project_name
+        version = metadata_files.metadata.version
+        try:
+            # type: (MetadataFiles) -> DistMetadata
+            return cls(
+                files=metadata_files,
+                project_name=project_name,
+                version=version,
+                requires_dists=tuple(requires_dists(metadata_files)),
+                requires_python=requires_python(metadata_files),
+            )
+        except RequirementParseError as e:
+            error_message = f"Error parsing requirements for project {project_name} version {version}: {str(e)}"
```

Comment:
Similar to above, f-strings are a no-go, you'll need to switch to `"...".format(...)`.